staging/src/k8s.io/apimachinery/pkg/util/diff/diff_test.go

```diff
@@ -27,3 +27,25 @@ func TestStringDiff(t *testing.T) {
 		t.Errorf("diff returned %v", diff)
 	}
 }
+
+func TestLegacyDiff(t *testing.T) {
+	equalResult := legacyDiff(true, true)
+	diffResult := legacyDiff(true, "string")
+
+	if equalResult != "" {
+		t.Errorf("two param type are same, result should be empty, but now the result is: %v", equalResult)
+	}
+
+	if diffResult == "" {
```

Comment:
Can we check the value of `diffResult` in some additional way (beyond seeing if it's non-empty)? I think I'm a little concerned about the output being non-empty but still incorrect.

Reply:
I have modified it. Thanks for your review.